sw/airborne/modules/decawave/decawave_serial_communication.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Serial_Communication.c
+ *
+ *  Created on: Jul 25, 2017
+ *      Author: Steven van der Helm
+ */
+
+/*
+ * Copyright (C) C. DW
+ *
+ * This file is part of paparazzi
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+/**
+ * @file "modules/decawave_serial_communication.h"
+ * @author S. vd H, C. DW
+ *
+  Decawave DWM1000 module serial communication for use in anchorless network where the UWB modules are attached to MAVs and need to communicate 
+  on-board values with each-other (for purposes such as relative localization, co-ordination, or collision avoidance).
+  This module must be used together with the Decawave DWM1000 running the appropriate Serial Communication code, which can be flashed on the Arduino board.
+  The arduino library can be found at:
+    https://github.com/StevenH2812/arduino-dw1000/tree/UWB_localization_v1.0
+  The example file to flash the Arduino Micro can be found in
+    examples/UWB_localization_v1_0/UWB_localization_v1_0.ino
+ */
+
+#include "decawave_serial_communication.h"
+#include "subsystems/datalink/telemetry.h"
+#include "subsystems/radio_control.h"
+#include "state.h"
+#include "pprzlink/messages.h"
+#include "subsystems/datalink/downlink.h"
+#include "mcu_periph/uart.h"
+#include <stdio.h>
+#include "subsystems/abi.h"
+
+PRINT_CONFIG_VAR(SERIAL_UART)
+PRINT_CONFIG_VAR(SERIAL_BAUD)
+
+// define coms link for stereocam
+#define SERIAL_PORT   (&((SERIAL_UART).device))
+struct link_device *xdev = SERIAL_PORT;
+
+#define SerialGetch() SERIAL_PORT ->get_byte(SERIAL_PORT->periph)
```

Comment:
Not sure these macros are a big help. Sometimes you use `SERIAL_PORT`, sometimes `xdev` (which is not a very explicit name). Would be better to be consistent.